src/components/Calendar/utils.js

```diff
@@ -63,95 +69,190 @@ export const getCountOfDayInNextMonth = (countOfWeek, countOfDay, countOfDaysInP
 export const getNextMonth = (date) => new Date(date.getFullYear(), date.getMonth() + 1, 1);
 
 /**
  * Convert Date for comparison
  * @param {Date} date - current date
  * @returns {number} - converted date
-*/
+ */
 export const convertDate = (date) => new Date(date).getTime();
+
+/**
+ * Get index of current Month
+ * @param {number} month - current month
+ * @param {Array} monthNames - current month
+ * @returns {number} - index current month
+ */
+export const getMonthIndex = (month, monthNames) => monthNames.indexOf(month);
+
+/**
+ * Get date from year select
+ * @param {Date} date - current date
+ * @param {number} year - selecting year
+ * @returns {Date} - selecting date
+ */
+export const getDateFromMonthSelect = (date, month) => new Date(date.getFullYear(), month, 1);
+
+/**
+ * Get date from year select
+ * @param {Date} date - current date
+ * @param {number} month - selecting month
+ * @returns {Date} - selecting date
+ */
+export const getDateFromYearSelect = (date, year) => new Date(year, date.getMonth(), 1);
+
+/**
+ * Get index of next month
+ * @param {number} month - index current month
+ * @returns {number} - index next month
+ */
+export const getIndexOfNextMonth = (month) => (month === 11 ? 0 : month + 1);
+
+/**
+ * Get index of prev month
+ * @param {number} month - index current month
+ * @returns {number} - index prev month
+ */
+export const getIndexOfPrevMonth = (month) => (month === 0 ? 11 : month - 1);
+
+/**
+ * Get value of prev year
+ * @param {number} month - index current month
+ * @param {number} year - index current year
+ * @returns {number} - index prev year
+ */
+export const getValueOfPrevYear = (month, year) => (month === 0 ? year - 1 : year);
+
+/**
+ * Get value of next year
+ * @param {number} month - index current month
+ * @param {number} year - index current year
+ * @returns {number} - index next year
+ */
+export const getValueOfNextYear = (month, year) => (month === 11 ? year + 1 : year);
+
+/**
+ * Disabled PrevYear button
+ * @param {number} month - index current month
+ * @param {number} year - index current year
+ * @param {number} startingYear - starting year
+ * @returns {number} - index next year
+ */
+export const disabledPrevYearButton = (month, year, startingYear) =>
+  month === 0 && year <= startingYear && 'disabled';
+
+/**
+ * Disabled NextYear button
+ * @param {number} month - index current month
+ * @param {number} year - index current year
+ * @param {number} endingYear - ending year
+ * @returns {number} - index next year
+ */
+export const disabledNextYearButton = (month, year, endingYear) =>
+  month === 11 && year >= endingYear && 'disabled';
+
+/**
+ * Set years for select
+ * @param {number} startingYear - starting year
+ * @param {number} endingYear - ending year
+ * @returns {Array} - array of years
+ */
+export const setYearValues = (startingYear, endingYear) => {
+  const yearsArray = [];
+  for (let i = startingYear; i <= endingYear; i++) { // eslint-disable-line
```

Comment:
I found more interesting way to do it:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#sequence_generator_range